File: gpp-api/src/main/java/com/develop/gpp/domain/entity/PecasEstoqueModel.java
package com.develop.gpp.domain.entity;

import javax.persistence.*;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "pecas_estoque")
public class PecasEstoqueModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idPecaEstoque;

    @ManyToOne
    @JoinColumn(name = "idPeca")
    private PecaModel peca;

    @Column(nullable = false)
    private Integer saldoDisponivel;

    @Column(nullable = false)
    private Integer saldoReservado;

    private String fornecedor;

    private String endereco;

    private Integer quantidadeMinima;

    private Integer quantidadeMaxima;

    private Integer quantidadeIdeal;


}

File: angular-gpp/src/app/models/PecasEstoqueModel.ts
import { PecaModel } from "./PecaModel";

export class PecasEstoqueModel {
  idPecaEstoque: number;
  peca: PecaModel;
  saldoDisponivel: number;
  saldoReservado: number;
  fornecedor?: string;
  endereco?: string;
  quantidadeMinima?: number;
  quantidadeMaxima?: number;
  quantidadeIdeal?: number;

  constructor(
    idPecaEstoque: number,
    peca: PecaModel,
    saldoDisponivel: number,
    saldoReservado: number,
    fornecedor?: string,
    endereco?: string,
    quantidadeMinima?: number,
    quantidadeMaxima?: number,
    quantidadeIdeal?: number
  ) {
    this.idPecaEstoque = idPecaEstoque;
    this.peca = peca;
    this.saldoDisponivel = saldoDisponivel;
    this.saldoReservado = saldoReservado;
    this.fornecedor = fornecedor;
    this.endereco = endereco;
    this.quantidadeMinima = quantidadeMinima;
    this.quantidadeMaxima = quantidadeMaxima;
    this.quantidadeIdeal = quantidadeIdeal;
  }
}
